qemu/tests/qemu_guest_agent.py

```diff
@@ -935,24 +1037,30 @@ def setup(self, test, params, env):
             vm = self.env.get_vm(params["main_vm"])
             vm.verify_alive()
             self.vm = vm
-            session = self._get_session(params, self.vm)
+        session = self._get_session(params, self.vm)
 
-            if self._check_ga_pkg(session, params.get("gagent_pkg_check_cmd")):
-                logging.info("qemu-ga is already installed.")
-            else:
-                logging.info("qemu-ga is not installed.")
-                self.setup_gagent_in_host(session, params, self.vm)
-                self.gagent_install(session, self.vm, *[params.get("gagent_install_cmd")])
+        if self._check_ga_pkg(session, params.get("gagent_pkg_check_cmd")):
+            logging.info("qemu-ga is already installed.")
+        else:
+            logging.info("qemu-ga is not installed.")
+            self.setup_gagent_in_host(session, params, self.vm)
+            self.gagent_install(session, self.vm, *[params.get("gagent_install_cmd")])
 
-            if self._check_ga_service(session, params.get("gagent_status_cmd")):
-                logging.info("qemu-ga service is already running.")
-            else:
-                logging.info("qemu-ga service is not running.")
-                self.gagent_start(session, self.vm)
+        if self._check_ga_service(session, params.get("gagent_status_cmd")):
```

Comment:
can we have variables and set it in `QemuGuestAgentTest()` initialization for qemu-ga package availability and status of it's service, so that we can avoid the checking multiple times?

Reply:
please help to remove unwanted comment from L1024

Reply:
After I re-read the whole code, I find the package check only appears in setup method, it isn't checked multiple times. And the check is must.

Reply:
I will add @ default and remove unwanted comment.

Reply:
L243 and L1049 we call self._check_ga_service() L237 and L1042 call self._check_ga_service(), as you say the check is must, we can check it once during __init__ of QemuGuestAgentTest() and set a boolean variable.
we can use the variable instead of calling the function at multiple places.

Reply:
Yes, I call it one time for windows and linux respectively. If I put the check in __init__ , then I also need to put L230-L235 in it.  And a if judgement is also needed, because only if the package is installed, then we can check whether the service is started. These operations are in front of self.initialize. So I don't think it is proper to put it in __int__.

Reply:
Agree with @balamuruhans , it could be initialized at the beginning